src/test/ui/symbol-names/const-generics-str-demangling.stderr

```diff
@@ -1,3 +1,4 @@
+WARN rustc_symbol_mangling computing the symbol name for an unexpected definition: DefId(0:6 ~ c[HASH]::{impl#0})
```

Comment:
Were do these warnings come from?

Reply:
from my changes to `rustc_symbol_mangling/src/lib.rs`
```rust
    let attrs = if tcx.def_kind(def_id).has_codegen_attrs() {
        tcx.codegen_fn_attrs(def_id)
    } else {
        warn!("computing the symbol name for an unexpected definition: {def_id:?}");
        CodegenFnAttrs::EMPTY
    };
```

Considering that this `warn!` only gets triggered in tests which force this by adding an attribute to impls, I think its sensible. Though maybe it would be even better to only allow `#[rustc_symbol_name]` on functions and statics :shrug: Don't have a very strong opinion here, so removing that `warn` would also be acceptable

Reply:
My worry was that this warn would appear to users. Since it is a `rustc_` attribute, this is not really a big deal. Still, I'd rather keep clean of this kind of output.

Reply:
If this warn appears to users that would always indicate either a bug or a "misuse" of the attribute, i.e. getting the symbol name of something that isn't a symbol